lib/registry/registry.go

```diff
@@ -71,11 +73,35 @@ func (r *registryImpl) CreateWorker(
 		return nil, errors.Trace(err)
 	}
 
-	worker, err := factory.NewWorker(ctx, workerID, masterID, config)
+	impl, err := factory.NewWorkerImpl(ctx, workerID, masterID, config)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return worker, nil
+
+	if implHasMember(impl, getTypeNameOfVarPtr(new(lib.BaseWorker))) {
```

Comment:
What about creating a global variable and reusing it
`var nameOfBaseWorker string = getTypeNameOfVarPtr(new(lib.BaseWorker))`, ditto for `lib.BaseJobMaster`.